lib/intercom/request.rb

```diff
@@ -75,8 +75,12 @@ def execute(target_base_url=nil, username:, secret: nil)
           rescue Intercom::RateLimitExceeded => e
             if @handle_rate_limit
               seconds_to_retry = (@rate_limit_details[:reset_at] - Time.now.utc).ceil
-              sleep seconds_to_retry unless seconds_to_retry < 0
-              retry unless (retries -=1).zero?
+              if (retries-=1) < 0
+                raise Intercom::RateLimitExceeded.new('armbar from the back')
```

Comment:
Can we display a clearer error message here? Sometimes who checks these logs aren't the same people that install the library, so we should add some context here.

Maybe something like: `Intercom rate limit exceeded - trying again in #{seconds_to_retry}`